### PR TITLE
Provide cmake install targets and install filesystembase.h

### PR DESCRIPTION
**Pull request by TheOneRing**

### CODE REVIEW

File: src/csync/CMakeLists.txt [hunk 1]
```diff
@@ -25,10 +25,13 @@ configure_file(csync_version.h.in ${CMAKE_CURRENT_BINARY_DIR}/csync_version.h)
...
 add_library(csync SHARED ${common_SOURCES} ${csync_SRCS})
 
+add_library(ownCloud::CSync ALIAS csync)
+set_target_properties(csync PROPERTIES EXPORT_NAME CSync)
+
 target_include_directories(csync
     PRIVATE ${CMAKE_CURRENT_SOURCE_DIR}/std
     # include root source dir in order to make src/common available to the targets
-    PUBLIC ${PROJECT_SOURCE_DIR}/src
+    PUBLIC $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/src>
```
Comment (fmoc) [inline src/csync/CMakeLists.txt:34]: I think you should include a line like `$<INSTALL_INTERFACE:include>`, too.

File: src/csync/CMakeLists.txt [hunk 2]
```diff
@@ -91,7 +96,16 @@ set_target_properties(
       SOVERSION
         "0"
 )
-INSTALL(TARGETS csync ${KDE_INSTALL_TARGETS_DEFAULT_ARGS})
+
+INSTALL(
+    FILES
+    ${CMAKE_CURRENT_BINARY_DIR}/ocsynclib.h
```
Comment (fmoc) [inline src/csync/CMakeLists.txt:102]: Indentation is off.

File: src/csync/CMakeLists.txt [hunk 3]
```diff
@@ -91,7 +96,16 @@ set_target_properties(
...
+        ../common/filesystembase.h
+        ../common/utility.h
+    DESTINATION ${INCLUDE_INSTALL_DIR}/ownCloud/common
```
Comment (fmoc) [inline src/csync/CMakeLists.txt:105]: Code style wise, the argument should be moved to a new line.

File: src/libsync/CMakeLists.txt [hunk 4]
```diff
@@ -68,22 +68,26 @@ else()
...
+    DESTINATION ${INCLUDE_INSTALL_DIR}/ownCloud/libsync
 )
 
 add_library(libsync SHARED ${libsync_SRCS})
-
-target_link_libraries(libsync PUBLIC
-    csync
-    Qt5::Core Qt5::Network
+add_library(ownCloud::SyncCore ALIAS libsync)
+set_target_properties(libsync PROPERTIES EXPORT_NAME SyncCore)
```
Comment (fmoc) [inline src/libsync/CMakeLists.txt:81]: Perhaps just rename `libsync` entirely?

File: src/libsync/CMakeLists.txt [hunk 5]
```diff
@@ -96,7 +100,7 @@ target_compile_definitions(libsync
             QT_NO_FOREACH
 )
 
-target_link_libraries(libsync PUBLIC OpenAPI::LibreGraphAPI)
+target_link_libraries(libsync PUBLIC $<BUILD_INTERFACE:OpenAPI::LibreGraphAPI>)
```
Comment (fmoc) [inline src/libsync/CMakeLists.txt:103]: See above, you probably should add an install interface include.

**Review by fmoc — CHANGES_REQUESTED** [review]

File: src/libsync/CMakeLists.txt (hunk 4 above)
Comment (TheOneRing) [reply src/libsync/CMakeLists.txt:81]: We can now use ownCloud::SyncCore in our code

File: src/libsync/CMakeLists.txt (hunk 5 above)
Comment (TheOneRing) [reply src/libsync/CMakeLists.txt:103]: Its not ready for use yet, and the 3 headers  we install don't depend on it.

Comment (TheOneRing) [reply src/libsync/CMakeLists.txt:103]: Might be a bit hacky, its basically a PRIVATE lib and I was to lazy to explicitly link the ui to it..

File: src/libsync/CMakeLists.txt (hunk 4 above)
Comment (fmoc) [reply src/libsync/CMakeLists.txt:81]: What I mean is, we probably should just rename directories etc. in our code base, too, and use one name everywhere.

Comment (fmoc) [reply src/libsync/CMakeLists.txt:81]: If this is the plan, a TODO should be left above this code.

**Review by fmoc — COMMENTED** [review]
Let's just consider this a "first start" then.

**Review by fmoc — APPROVED** [review]

**Pull request merged**